api/v1/report.proto

```diff
@@ -18,131 +18,18 @@ package istio.mixer.v1;
-
-  // The value. The type of value used in the request must
-  // agree with the MetricDescriptor, otherwise
-  // the MetricValue is rejected.
-  oneof value {
-    // A boolean value.
-    bool bool_value = 5;
-
-    // A signed 64-bit integer value.
-    int64 int64_value = 6;
-
-    // A double precision floating point value.
-    double double_value = 7;
-
-    // A text string value.
-    string string_value = 8;
-  }
-}
-
-message LogEntry {
-  // The severity of the event described in a log entry, expressed as one of the
-  // standard severity levels listed below.  For your reference, the levels are
-  // assigned the listed numeric values. The effect of using numeric values other
-  // than those listed is undefined.
-  //
-  // If you are writing log entries, you should map other severity encodings to
-  // one of these standard levels. For example, you might map all of Java's FINE,
-  // FINER, and FINEST levels to `LogSeverity.DEBUG`. You can preserve the
-  // original severity level in the log entry payload if you wish.
-  enum LogSeverity {
-    // (0) The log entry has no assigned severity level.
-    DEFAULT = 0;
-
-    // (100) Debug or trace information.
-    DEBUG = 100;
-
-    // (200) Routine information, such as ongoing status or performance.
-    INFO = 200;
-
-    // (300) Normal but significant events, such as start up, shut down, or
-    // a configuration change.
-    NOTICE = 300;
-
-    // (400) Warning events might cause problems.
-    WARNING = 400;
-
-    // (500) Error events are likely to cause problems.
-    ERROR = 500;
-
-    // (600) Critical events cause more severe problems or outages.
-    CRITICAL = 600;
-
-    // (700) A person must take an action immediately.
-    ALERT = 700;
-
-    // (800) One or more systems are unusable.
-    EMERGENCY = 800;
-  }
-
-  // The name of the LogEntryDescriptor providing the schema for this log entry.
-  string descriptor_name = 1;
-
-  // Destination log collection for this entry
-  string log_collection = 2;
-
-  // The time the event described by the log entry occurred.
-  google.protobuf.Timestamp timestamp = 3;
-
-  // The severity of the log entry. The default value is
-  // `LogSeverity.DEFAULT`.
-  LogSeverity severity = 4;
-
-  // The log entry payload, which can be one of multiple types.
-  oneof payload {
-    // The log entry payload, represented as a protocol buffer that is
-    // expressed as a JSON object. You can only pass `protoPayload`
-    // values that belong to a set of approved types.
-    google.protobuf.Any proto_payload = 5;
-
-    // The log entry payload, represented as a Unicode string (UTF-8).
-    string text_payload = 6;
-
-    // The log entry payload, represented as a structure that
-    // is expressed as a JSON object.
-    google.protobuf.Struct struct_payload = 7;
-  }
+  // The set of attributes associated with this request. This value represents
+  // a delta relative to the attributes currently known to the mixer within
```

Comment:
A couple of general thoughts.

1. For Check, we should only check once per minute for the same set of attributes. The server should return the set of actually used attributes. This would give us an infinitely scalable design, and also very easy to understand.

2. For Report, different backend requires different schema and has different restrictions. It is certain we need to report to different parties at the same time, billing, logging, monitoring, auditing, producer, consumer, auditor, network, health checker, load balancer, and more. How to design a usable and scalable solution is a bit unknown. We should study the existing systems.

We can model it as a set of attributes, and let mixer decide which attribute goes where. It would work well except the efficiency. We can start with this.